tdms/src/iterator.cpp

```diff
@@ -4004,8 +4004,7 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[]) {
             eh_vec[n][j][0] = E_s.xy[k][j][i] + E_s.xz[k][j][i];
             eh_vec[n][j][1] = 0.;
 
-            first_derivative(eh_vec[n], eh_vec[n], dk_h_y,
-                             N_h_y, H_s.zy.plan_f[n], H_s.zy.plan_f[n]);
+            first_derivative(eh_vec[n], eh_vec[n], dk_h_y, N_h_y, H_s.zy.plan_f[n], H_s.zy.plan_b[n]);

```

Comment:
A very very naive question: could such a thing have been covered by a unit test?

Reply:
definitely could be, but not at this moment given it's a part of a 4k line function 😞 – hopefully in the near future this can be broken up into more manageable chunks. only managed to get to the start of the main loop with my refactor